File: go/237.delete-node-in-a-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteNode(node *ListNode) {
    // since node will not be the tail
    // we can copy the next node's value to node
    // and delete the next node
    // O(1)
    node.Val = node.Next.Val
    node.Next = node.Next.Next
}

File: csharp/237.delete-node-in-a-linked-list.cs
/*
 * @lc app=leetcode id=237 lang=csharp
 *
 * [237] Delete Node in a Linked List
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution 
{
    public void DeleteNode(ListNode node) 
    {
        ListNode prev = null;
        while (node.next != null)
        {
            node.val = node.next.val;
            prev = node;
            node = node.next;
        }

        prev.next = null;
    }
}
// @lc code=end

